Getting started with Okteto Test (#750)
okteto/docs: src/content/testing/getting-started.mdx

---
title: Getting started with Test Containers

> description: In this section, we'll see how you caconfigure Test Containers for your application

Reviewer: ```suggestion
description: In this section, we'll see how you configure Test Containers for your application
```

sidebar_label: Getting Started
id: getting-started
---

This guide explains how to configure Test Containers for your first app to Okteto.

## Requirements

Before you start this tutorial, make sure you fulfill the following requirements:

- You have access to an Okteto instance. Otherwise, follow our [installation guide](get-started/install/index.mdx)
- You have [configured the Okteto CLI](get-started/install-okteto-cli.mdx) with your Okteto instance
- You have complited [our guide to deploy your first app to Okteto](get-started/deploy-your-app/index.mdx)

## What you will be building

This tutorial will guide you on how to test [Movies app](https://github.com/okteto/getting-started) using Okteto.

## Step 1: Deploy the Movies App

> Get a local version of the Go Sample App by executing the following commands:

Reviewer: Movies maybe?
```suggestion
Get a local version of the Movies Sample App by executing the following commands:
```

Reviewer 2: Yes, in the current docs we mostly use "Movies app" for the other section of the getting started. cc. @pchico83

```bash
git clone https://github.com/okteto/getting-started
cd getting-started
```

Switch to the `onboarded` branch:

```bash
git checkout onboarded
```

At the root of the directory, you'll find the `okteto.yaml` file.

> This file describes how to build, deploy and test the Movies App.

Reviewer 2: ```suggestion
This file describes how to build, deploy and test the Movies app.
```

Deploy your development environment by executing:

```bash
okteto deploy
```

```bash
 i  Using cindy @ okteto.example.com as context
 i  Running 'Helm Install'
Release "movies" has been upgraded. Happy Helming!
NAME: movies
LAST DEPLOYED: Tue Jun  4 15:37:11 2024
NAMESPACE: pchico83
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
Success! Your application will be available shortly.
 i  Endpoints available:
  - https://movies-cindy.okteto.example.com
 ✓  Development environment 'getting-started' successfully deployed
 ```

## Step 2: Execute the `frontend` unit tests in a Test Container

> The [test](reference/okteto-manifest.mdx#dev-object-optional) section in the Okteto Manifest defines how to test the Movies App.

Reviewer 2: nit. "app" is used in lower case in the rest of the docs...
```suggestion
The [test](reference/okteto-manifest.mdx#dev-object-optional) section in the Okteto Manifest defines how to test the Movies app.
```

 i  Executing test container 'frontend'
 i  Running 'yarn install'
yarn install v1.22.22
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 43.31s.
 ✓  Command 'yarn install' successfully executed
 i  Running 'yarn test'
yarn run v1.22.22
$ jest
PASS src/components/Hero.test.jsx
PASS src/components/TitleList.test.jsx
PASS src/components/Item.test.jsx
PASS src/components/App.test.jsx
PASS src/components/UserProfile.test.jsx
PASS src/components/ListToggle.test.jsx
Test Suites: 6 passed, 6 total
Tests:       20 passed, 20 total
Snapshots:   0 total
Time:        5.658 s
Ran all test suites.
Done in 6.59s.
 ✓  Command 'yarn test' successfully executed
 ✓  Test container 'frontend' passed 
```

If you rerun your Test Container:

```bash
okteto test frontend
```

```bash
 i  Using cindy @ okteto.example.com as context
 i  Executing test container 'frontend'
 i  Skipping test container 'frontend', CACHED
 ✓  Test container 'frontend' passed
```

You can see that Okteto skips the execution of the tests. This is because the content of the folder `context: frontend` didn't change.

> This is very convinient in monorepos, since Okteto won't rerun unit tests in the folder has no changes.

Reviewer 2: ```suggestion
This is very convenient in monorepos, since Okteto won't rerun unit tests in the folder has no changes.
```

Reviewer 2: Or maybe a better phrasing, something like:
```suggestion
This is especially convenient in monorepos because Okteto will only rerun unit tests in folders with changes, saving time and resources.
```
